src/basic/btrfs-util.c

```diff
@@ -917,6 +917,10 @@ int btrfs_resize_loopback_fd(int fd, uint64_t new_size, bool grow_only) {
         if (new_size < 16*1024*1024)
                 new_size = 16*1024*1024;
 
+        /* Linux VFS cannot handle file sizes > 8EiB, because off_t is signed 64bit */
+        if (new_size > INT64_MAX)
+                new_size = INT64_MAX;
+
```

Comment:
Well, we have two interfaces here that need attention.

One is the DBus interface, which takes an *unsigned* 64bit integer. We should keep this like it is, and only treat `(uint64_t) -1` (or `UINT64_MAX`, respectively) special for 'unlimited'.

The other side is the DBus call receiver side, however. And here, we are limited due to kernel APIs that take *signed* 64bit integers, so we cannot support any parameter value which is `> INT64_MAX`, and so any call with such a parameter should fail, and the error needs to be upchained to the DBus client. With the exception though that `UINT64_MAX` input needs to be translated to `(int64_t) -1` for the syscall.

Make sense?